integrations/spark/src/main/java/marquez/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -142,14 +171,20 @@ protected String getEventType(JobResult jobResult) {
   }
 
   protected RunFacet buildRunFacets(
-      LogicalPlanFacet logicalPlanFacet, ErrorFacet jobError, ParentRunFacet parentRunFacet) {
+          LogicalPlanFacet logicalPlanFacet,
+          ErrorFacet jobError,
+          ParentRunFacet parentRunFacet,
+          DatasetMetricFacet metricFacet) {
     Map<String, Object> additionalFacets = new HashMap<>();
     if (logicalPlanFacet != null) {
       additionalFacets.put("spark.logicalPlan", logicalPlanFacet);
     }
     if (jobError != null) {
       additionalFacets.put("spark.exception", jobError);
     }
+    if (metricFacet != null) {
+      additionalFacets.put("dataset.metrics", metricFacet);
```

Comment:
This facet should be an outputFacet on the output dataset rather than on the run.